SteamList/Controllers/GamesViewController.swift

```diff
@@ -53,8 +51,26 @@ extension GamesViewController: UITableViewDataSource {
     }
     
     
+     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+         navigate(tableView, indexPath)
+    }
+    
+    func navigate(_ tableView: UITableView,  _ indexPath: IndexPath) {
+         var gameObj = MockData.games[indexPath.row]
+             let cell = tableView.cellForRow(at: indexPath)  as! GameTableViewCell
+             if cell.cellIcon.imageView?.image == UIImage(systemName: "star") {
+                 gameObj.isFavorite = false
+             } else {
+                 gameObj.isFavorite = true
+             }
+         guard let gameDetailsViewController = self.storyboard?.instantiateViewController(identifier: "GameDetailsViewController",
+        creator: {coder -> GameDetailsViewController? in
+              GameDetailsViewController(coder: coder, game: gameObj)
+         }) else {return}
+        self.navigationController?.pushViewController(gameDetailsViewController, animated: true)
+    }
+    
```

Comment:
This can be pretty confusing to undestand and volatile if the id on storyboard gets changed somehow in the future. Suggested to refactor it a little bit to something like below:

```Swift
    guard let gameDetailsViewController = storyboard?.instantiateViewController(withIdentifier: "GameDetailsViewController") as? GameDetailsViewController else {
      assertionFailure("View controller with `GameDetailsViewController` identifier could not be instantiated from storyboard")
      return
    }

    gameDetailsViewController.game = game
    navigationController?.pushViewController(gameDetailsViewController, animated: true)
```

This way, if storyboard is not found with the given id, you will see a crash (only in debug mode) and easyly fix it
Also, it is much easier to understand without the trailing closure of of view controller initializer

Reply:
got it